clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1628,6 +1628,12 @@ scheduler_tmpl_t<RecordType, ReaderType>::record_schedule_segment(
     // We always use the current wall-clock time, as the time stored in the prior
     // next_record() call can be out of order across outputs and lead to deadlocks.
     uint64_t timestamp = get_time_micros();
+    if (type == schedule_record_t::IDLE &&
```

Comment:
Add a check to some existing test at scheduler_unit_tests.cpp?

Reply:
It will not be trivial as while some tests do construct synthetic schedule files none of them peer inside written schedule files, and that is the only way to tell as otherwise it is a performance change.  I will spend a little time to see how easy it is.

Reply:
I had to add a friend class to access the non-public schedule record type (those synthesized test ones are as-traced, not replay) but I did manage to add checks which fail without this fix.

Reply:
I think a separate issue on this may be worthwhile in case others have hit this; I know someone internally did see very large files.  Will file.